### PR TITLE
Add 32-bit build to AppVeyor

### PR DESCRIPTION
**Pull request by tkuchida**
See #902

TODO:
- [ ] Open issues to resolve known test failures on 32-bit Windows and remove them from the `OPENSIM_TESTS_TO_EXCLUDE` and `PYTHON_TESTS_TO_EXCLUDE` environment variables in appveyor.yml.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -44,7 +51,7 @@ install:
   - choco install swig --version 3.0.5 -y
   
   ## Install python-nose for python testing.
-  - C:\Python27-x64\Scripts\pip install nose
+  - %PYTHON_DIR%\Scripts\pip install nose
```
Comment (chrisdembia) [inline appveyor.yml]: Maybe put the whole path in quotes.

Comment (chrisdembia) [inline appveyor.yml]: `"%PYTHON_DIR%\Scripts\pip"`

Comment (tkuchida) [inline appveyor.yml]: Thanks. I found [this page](http://help.appveyor.com/discussions/problems/2869-appdata-causes-error-parsing-appveyoryml-while-scanning-for-the-next-token-find-character-that-cannot-start-any-token), which uses single-quotes, but I'll try double-quotes… must be twice as good.

File: appveyor.yml [hunk 2]
```diff
@@ -21,13 +21,20 @@ os: Visual Studio 2015
 
 platform: x64
 
+environment:
+  matrix:
+    - PYTHON_DIR: "C:\\Python27"
+      CMAKE_GENERATOR: "Visual Studio 14 2015"
+    - PYTHON_DIR: "C:\\Python27-x64"
+      CMAKE_GENERATOR: "Visual Studio 14 2015 Win64"
+
```
Comment (klshrinidhi) [inline appveyor.yml]: I think we need not specify a generator for CMake if we want to build 32-bit. It is the default generator chosen anyway with MSVC.
It is still good to keep it explicit so readers of this file can know what is being used.

**Comment by klshrinidhi** [comment]
All :x: :anguished:

File: appveyor.yml (hunk 2 above)
Comment (chrisdembia) [inline appveyor.yml]: > It is still good to keep it explicit so readers of this file can know what is being used.

I agree; the Appveyor machines may have multiple versions of MSVC also.

**Comment by tkuchida** [comment]
Need to get 32-bit version of Simbody built and uploaded first. Working on it [here](https://github.com/simbody/simbody/tree/appveyor-add-32bit).

**Comment by tkuchida** [comment]
Status:
- 64-bit build successful; all tests pass
- 32-bit build successful; testC3DFileAdapter and python_Release segfaulted (the latter seems to be related to testC3DFileAdapter also)

@chrisdembia - Can you tell whether [the segfaults](https://ci.appveyor.com/project/opensim-org/opensim-core/build/3120/job/9spibei8tx6mppnq) are due to a problem with the yml file? If not, perhaps this PR is ready for review and a separate issue should be raised for the test failures. Unfortunately, these failures are different from the ones reported in #784.

**Comment by chrisdembia** [comment]
We are unable to see the output of the failed tests because our appveyor build cuts off at 1 hour (well, 59min 59sec). This also means that there may be more tests failures than we currently can see. 

The issue may be a bug with BTK when building it as 32bit. The only potential issue with the yml file I think would have been if you did not properly set the generator for the superbuild, but you did. 

> perhaps this PR is ready for review and a separate issue should be raised for the test failures.

I agree; the C3D failure should be debugged locally, not in this PR. but I think you should also use CTEST to avoid running the tests with the known failures on the 32-bit build. [We already do this on travis to exclude tests that are known to fail in debug](https://github.com/opensim-org/opensim-core/blob/master/.travis.yml#L158); search for all occurrences of "TESTS_TO_EXCLUDE." I think it'd be cleaner to do this as an environment variable as part of `matrix` in `appveyor.yml`, instead of the messy way I chose to do it for travis.

**Comment by tkuchida** [comment]
Ready for review (AppVeyor timed out on one of the two x64 jobs). Perhaps wait for review/merge of [associated Simbody PR](https://github.com/simbody/simbody/pull/481) before merging this one.

**Comment by chrisdembia** [comment]
Looks good to me. Did you notice that the 32-bit build takes almost half the amount of time on AppVeyor? (33 minutes vs over an hour)? I wonder why that is.

**Comment by tkuchida** [comment]
> Did you notice that the 32-bit build takes almost half the amount of time on AppVeyor? (33 minutes vs over an hour)? I wonder why that is.

[The most recent AppVeyor build on master](https://ci.appveyor.com/project/opensim-org/opensim-core/branch/master) took 55½ minutes, so this PR hasn't affected the duration of the 64-bit build.

**Comment by chrisdembia** [comment]
> The most recent AppVeyor build on master took 55½ minutes, so this PR hasn't affected the duration of the 64-bit build.

I should have clarified. I know that the 64-bit build takes about an hour. That's why I was happily surprised to see the 32-bit build taking much less time. I don't understand why this is though.

**Comment by chrisdembia** [comment]
Looking at the appveyor build log, the speedup mostly comes from a reduced amount of time required for compiling (the tests take roughly the same amount of time). If you hover over the line number in the appveyor log, the hovertext tells you the timestamp for that line.

**Comment by tkuchida** [comment]
There are 17 more build warnings on 64-bit compared to 32-bit (261 vs. 244).

**Comment by chrisdembia** [comment]
Oh weird...

**Comment by klshrinidhi** [comment]
I think this PR should wait and see if PR #916 fixes the 32-bit `C3DFileAdapter` failure issue. It would be good to not exclude the test.

**Comment by tkuchida** [comment]
@chrisdembia Did you intend to merge this? I think it should at least wait until the corresponding PR has been merged in Simbody.

**Comment by klshrinidhi** [comment]
Remember to nuke ( :fire: ) the branch when you are done.

**Comment by chrisdembia** [comment]
No I did not intend to merge this. And I didn't, at least not manually. GitHub has counted this as merged when I merged #916. However, the changes are reverted (on master).

**Comment by chrisdembia** [comment]
The reason GitHub counted this as merged is that @klshrinidhi merged your branch into his branch (at my request). So I think one way to proceed is to create a new branch off of `appveyor-add-32bit`, and create a new PR for that?

**Pull request merged**